src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -386,21 +404,39 @@ export default {
 			}
 		},
 
+		token() {
+			return this.$store.getters.getToken()
+		},
+
+		getFilterCaption() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'Unread mentions')
+				: t('spreed', 'Unread messages')
+		},
+
+		messageDisplay() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'You have no unread mentions in your inbox')
+				: t('spreed', 'You have no unread messages in your inbox')
+		},
+
 		filteredConversationsList() {
 			if (this.isFocused) {
 				return this.conversationsList
 			}
-
-			if (this.isFiltered === 'unread') {
-				return this.conversationsList.filter(conversation => conversation.unreadMessages > 0
-				|| conversation.hasCall || conversation.token === this.$store.getters.getToken())
-			}
-
-			if (this.isFiltered === 'mentions') {
-				return this.conversationsList.filter(conversation => conversation.unreadMention
-				|| conversation.hasCall
-				|| (conversation.unreadMessages > 0 && (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER))
-				|| conversation.token === this.$store.getters.getToken())
+			// applying filters
+			if (this.isFiltered) {
+				const filteredConversations = this.conversationsList.filter((conversation) => this.filterFunction(conversation)
+					|| conversation.hasCall
+					|| conversation.token === this.token)
+
+				// return empty if it only includes the current conversation without any flags
+				const isNewFilter = this.isPreviousFilter() !== this.isFiltered
+				const hasCurrentConversationOnly = filteredConversations.length === 1
```

Comment:
`=== 1` won't work if you have current conversation + conversation in call.
We may count inside of `this.conversationsList.filter()` how many conversations are truthy for `this.filterFunction`, and then check that number (if at least 1 unread won't be there, show `[ ]`)